File: 1313-decompress-run-length-encoded-list/1313-decompress-run-length-encoded-list.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int total = 0;
        for (int i = 0; i < nums.length;i+=2){
            total += nums[i];
        }
        int[] arr = new int[total];
        int k = 0;
        for (int i = 0; i < nums.length;i+=2){
            int value = nums[i+1];
            int freq = nums[i];
            for (int j = 0; j < freq;j++){
                arr[k] = value;
                k++;
            }
        }
        return arr;  
    }
}

File: 1313-decompress-run-length-encoded-list/1313-decompress-run-length-encoded-list.py
class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        ans = []
        for i in range(0,len(nums)-1,2): # iterates from 0th to last 2nd element of nums taking 2 steps at a time
            ans += ([nums[i+1]] * nums[i]) # add [value] times freq to ans list
        return ans